packages/push/examples/smallest/main.rs

```diff
@@ -0,0 +1,215 @@
+pub mod args;
+
+use std::ops::Not;
+
+use anyhow::{ensure, Result};
+use clap::Parser;
+use ec_core::{
+    distributions::{collection::ConvertToCollectionGenerator, conversion::IntoDistribution},
+    generation::Generation,
+    individual::{ec::WithScorer, scorer::FnScorer},
+    operator::{
+        genome_extractor::GenomeExtractor,
+        genome_scorer::GenomeScorer,
+        mutator::Mutate,
+        selector::{best::Best, lexicase::Lexicase, Select, Selector},
+        Composable,
+    },
+    test_results::{self, TestResults},
+};
+use ec_linear::mutator::umad::Umad;
+use push::{
+    evaluation::cases::{Case, Cases, WithTargetFn},
+    genome::plushy::{GeneGenerator, Plushy},
+    instruction::{
+        variable_name::VariableName, BoolInstruction, ExecInstruction, IntInstruction,
+        PushInstruction,
+    },
+    push_vm::{program::PushProgram, push_state::PushState, stack::StackError, HasStack, State},
+};
+use rand::{distributions::Distribution, rngs::ThreadRng, thread_rng, RngCore};
+use strum::IntoEnumIterator;
+
+use crate::args::{Args, RunModel};
+
+// An input for this problem is a tuple of four `i64`s.
+#[derive(Copy, Clone)]
+struct Input(i64, i64, i64, i64);
+// An output for this problem is an `i64`.
+struct Output(i64);
```

Comment:
And then here 
```suggestion
type Output =i64;
```

Reply:
This avoids wrapping in places (like in the score fn below) and should make the code more concise

Reply:
I chose to keep the `structs` for `Input` and `Output`, in part so that the output `i64` couldn't be confused with other kinds of `i64`s (like inputs).